crates/sui-faucet/src/faucet/simple_faucet.rs

```diff
@@ -430,7 +430,13 @@ impl SimpleFaucet {
         let client = self.wallet.get_client().await?;
         client
             .transaction_builder()
-            .pay_sui(signer, vec![coin_id], recipients, amounts.to_vec(), budget)
+            .pay_sui_legacy_for_faucet_do_not_use_will_be_removed(
```

Comment:
@longbowlu @amnn I had to revert the changes for faucet as it was a bit too complicated for one PR. I ran into some issues and could use some help. I'll make another PR shortly